packages/admin/admin-generator/src/commands/generate/config/transformConfig.ts

```diff
@@ -0,0 +1,611 @@
+        return (context) => {
+            const visit = (node: ts.Node, path: string): ts.Node => {
+                if (ts.isArrowFunction(node)) {
+                    if (supportedInlineCodePaths.includes(path)) {
+                        let code = node.getText();
+                        if (code.endsWith(",")) code = code.slice(0, -1); // for some unknown reason node can contain the trailing comma
+                        const imports = findUsedImports(node.body, importedIdentifiers); //find all imports used in the function body
+                        // replace inline code with { code, imports } object
+                        return ts.factory.createObjectLiteralExpression(
+                            [
+                                ts.factory.createPropertyAssignment("code", ts.factory.createStringLiteral(code)),
+                                ts.factory.createPropertyAssignment(
+                                    "imports",
+                                    ts.factory.createArrayLiteralExpression(
+                                        imports.map((imprt) => {
+                                            return ts.factory.createObjectLiteralExpression([
+                                                ts.factory.createPropertyAssignment("name", ts.factory.createStringLiteral(imprt.name)),
+                                                ts.factory.createPropertyAssignment("import", ts.factory.createStringLiteral(imprt.import)),
+                                            ]);
+                                        }),
+                                    ),
+                                ),
+                            ],
+                            true,
+                        );
+                    } else {
+                        throw new Error(`Inline Function is not allowed here and calling the function is not supported: ${path}`);
+                    }
+                } else if (ts.isIdentifier(node)) {
+                    const imported = importedIdentifiers.get(node.text);
+                    if (imported) {
+                        if (supportedImportPaths.includes(path)) {
+                            // replace imported identifier with { name, import } object
+                            return ts.factory.createObjectLiteralExpression(
+                                [
+                                    ts.factory.createPropertyAssignment("name", ts.factory.createStringLiteral(node.text)),
+                                    ts.factory.createPropertyAssignment("import", ts.factory.createStringLiteral(imported.import)),
+                                ],
+                                true,
+                            );
+                        } else {
+                            throw new Error(`Following the import is not supported: ${path} ${node.text}`);
+                        }
+                    }
+                }
+
+                const transformKinds = [
```

Comment:
Do we need this long comment?

Reply:
no, removed [9824152](https://github.com/vivid-planet/comet/pull/3781/commits/9824152d25e67a9c6839b07fd6bc294e8a71e469)